### PR TITLE
Refactor page experience check to run cache test separately

### PR DESCRIPTION
**Pull request by tharders**
Cache check is not run if page is invalid amp.

This is also the basis for the last missing conditions in the statusBanner logic.
Localization of the texts will be done in a separate PR.

### CODE REVIEW

File: pixi/src/checks/PageExperienceCacheCheck.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        this.addImprovement(cacheLabData, originLabData, 'fid');
+      }
+    }
+    return cacheResult;
+  }
+
+  addImprovement(cacheMetric, originMetric, key) {
+    const cacheData = cacheMetric[key].data;
+    const originData = originMetric[key].data;
+    cacheData.improvement = originData.numericValue - cacheData.numericValue;
```
Comment (caroqliu) [inline pixi/src/checks/PageExperienceCacheCheck.js:59]:
```suggestion
    cacheData.improvement = Math.max(0, originData.numericValue - cacheData.numericValue);
```

File: pixi/src/checks/PageExperienceCacheCheck.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  async run(originUrl, originPageExperiencePromise) {
+    const cacheUrl = await ampToolboxCacheUrl.createCacheUrl(
+      AMP_PROJECT_CDN_URL,
+      originUrl
+    );
+
+    const cacheResult = await this.pageExperienceCheck.run(cacheUrl);
+    const originResult = await originPageExperiencePromise;
+    if (cacheResult.data && originResult.data) {
+      const cacheFieldData = cacheResult.data.pageExperience.fieldData;
```
Comment (caroqliu) [inline pixi/src/checks/PageExperienceCacheCheck.js:38]: Do we know that if `cacheResult.data` is defined, `cacheResult.data.pageExperience` will also be defined?

**Review by caroqliu — APPROVED** [review]

File: pixi/src/checks/PageExperienceCacheCheck.js (hunk 2 above)
Comment (tharders) [reply pixi/src/checks/PageExperienceCacheCheck.js:38]: Yes, the PageExperienceCheck returns either `error`, or `data` with `pageExperience`

File: pixi/src/checks/PageExperienceCacheCheck.js (hunk 1 above)
Comment (tharders) [reply pixi/src/checks/PageExperienceCacheCheck.js]: Done

File: pixi/src/checks/PageExperienceCacheCheck.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      const originLabData = originResult.data.pageExperience.labData;
+      if (cacheFieldData && originFieldData) {
+        this.addImprovement(cacheFieldData, originFieldData, 'lcp');
+        this.addImprovement(cacheFieldData, originFieldData, 'cls');
+        this.addImprovement(cacheFieldData, originFieldData, 'fid');
+      }
+      if (cacheLabData && originLabData) {
+        this.addImprovement(cacheLabData, originLabData, 'lcp');
+        this.addImprovement(cacheLabData, originLabData, 'cls');
+        this.addImprovement(cacheLabData, originLabData, 'fid');
```
Comment (caroqliu) [inline pixi/src/checks/PageExperienceCacheCheck.js:50]: Is this read at all from the response? Missed this, but the metric for lab data is "tbt" in lieu of "fid".

Comment (tharders) [reply pixi/src/checks/PageExperienceCacheCheck.js:50]: Currently the check still returns the "tbt" value as "fid", so this should work.

**Pull request merged**